joinrun/src/test/scala/code/chymyst/test/Patterns02Spec.scala

```diff
@@ -0,0 +1,173 @@
+package code.chymyst.test
+
+import java.util.concurrent.ConcurrentLinkedQueue
+
+import code.chymyst.jc._
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
+
+import scala.collection.JavaConverters.asScalaIteratorConverter
+import scala.language.postfixOps
+
+class Patterns02Spec extends FlatSpec with Matchers with BeforeAndAfterEach {
+
+  var tp: Pool = _
+
+  override def beforeEach(): Unit = {
+    tp = new SmartPool(4)
+  }
+
+  override def afterEach(): Unit = {
+    tp.shutdownNow()
+  }
+
+  behavior of "smokers"
+
+  it should "implement smokers" in {
+    val supplyLineSize = 10
+
+    def smokingBreak(): Unit = Thread.sleep(math.floor(scala.util.Random.nextDouble * 20.0 + 2.0).toLong)
+
+    case class ShippedInventory(tobacco: Int, paper: Int, matches: Int)
+    // this data is only to demonstrate effects of randomization on the supply chain and make content of logFile more interesting.
+    // strictly speaking all we need to keep track of is inventory. Example would work if pusher molecule value would carry Unit values instead.
+
+    val pusher = m[ShippedInventory]
+    // pusher means drug dealer, in classic Comp Sci, we'd call this producer or publisher.
+    val count = m[Int]
+    val Keith = new E("Keith obtained tobacco and matches to get his fix")
+    // makes for more vivid tracing, could be plainly m[Unit]
+    val Slash = new E("Slash obtained tobacco and matches to get his fix")
+    // same
+    val Jimi = new E("Jimi obtained tobacco and matches to get his fix") // same
+
+    val tobacco = m[ShippedInventory]
```

Comment:
visual layout of lines 43-48 is not right. We should get rid of the "//same" comments that don't add much. I'd prefer lines 43 with comment @44 on same line.